apps/api_web/lib/api_web/views/vehicle_view.ex

```diff
@@ -84,4 +85,21 @@ defmodule ApiWeb.VehicleView do
   defp backwards_compatible_attributes(attributes, _, _) do
     attributes
   end
+
+  defp encode_carriages(%{carriages: nil} = vehicle) do
+    vehicle
+  end
+
+  defp encode_carriages(vehicle) do
+    Map.put(vehicle, :carriages, Enum.map(vehicle.carriages, &encode_carriage/1))
+  end
+
+  defp encode_carriage(carraige) do
+    %{
+      label: carraige.label,
+      carriage_sequence: carraige.carriage_sequence,
+      occupancy_status: occupancy_status(carraige, nil),
```

Comment:
suggestion(non-blocking): rather than passing `nil` in for the conn, what about making `occupancy_status/1` be the result of the macro, and`occupancy_status/2` call `occupancy_status/1` without the conn?

```elixir
for status <- [...]
  def occupancy_status(unquote(status)), do: ...
end

def occupancy_status(vehicle, _conn), do: occupancy_status(vehicle.occupancy_status)

defp encode_carriage(carriage) do
  %{
    ...,
    occupancy_status: occupancy_status(carriage.occupancy_status)
  }
end
```

Reply:
Changed. It looks like what you describe is what JaSerializer actually does, defining `occupancy_status/1` and having `occupancy_status/2` call `/1`. I'm not sure why other views use the `/2` other than that's what the examples in the JaSerializer documentation do.